allauth/account/views.py

```diff
@@ -654,6 +654,12 @@ def logout(self):
                                   messages.SUCCESS,
                                   'account/messages/logged_out.txt')
         auth_logout(self.request)
+        signal_kwargs = {}
+        signals.user_logged_out.send(sender=self.request.user.__class__,
+                                    request=self.request,
+                                    user=self.request.user,
```

Comment:
`auth_logout` sets `request.user` to the anonymous user... so we lose the user that logged out. We need to preserve the user before `auth_logout` and use that here.